### PR TITLE
core: review usages of (*intero.Context).BlockHeight method

### PR DESCRIPTION
**Pull request by AnnaShaleva**
This method returns persisted block height and doesn't take into account persisting block height. Some of the callers of this method relay on the wrong assumption that BlockHeight() returns persisting block index.

Fix improper usages of this method and adjust tests. Ref. https://github.com/neo-project/neo/blob/61a066583eef53c8eb7b770dcc3deb0305ab11db/src/Neo/SmartContract/ApplicationEngine.cs#L634.

### CODE REVIEW

File: pkg/core/native/notary.go [hunk 1]
```diff
@@ -272,7 +272,7 @@ func (n *Notary) lockDepositUntil(ic *interop.Context, args []stackitem.Item) st
 		return stackitem.NewBool(false)
 	}
 	till := toUint32(args[1])
-	if till < ic.BlockHeight() {
+	if till < ic.BlockHeight() { // @roman-khimov, is it OK to lock deposit till the current persisting block?
```
Comment (AnnaShaleva) [inline pkg/core/native/notary.go:275]: We need to decide about this one. It's not critical to have it as it is now.

Comment (roman-khimov) [reply pkg/core/native/notary.go:275]: Yes, the other way to look at it is
https://github.com/nspcc-dev/neo-go/blob/f10d2d2be7de5927f532a177cc09f63877b121e5/pkg/core/native/notary.go#L311-L313
If you deposit, but can withdraw immediately the deposit is useless.

Comment (AnnaShaleva) [reply pkg/core/native/notary.go]: Good, I've fixed this in couple of places and added a comment to `withdraw`. Could you check it?

File: pkg/core/interop/context.go [hunk 2]
```diff
@@ -371,7 +371,9 @@ func (ic *Context) Exec() error {
 	return ic.VM.Run()
 }
 
-// BlockHeight returns current block height got from Context's block if it's set.
+// BlockHeight returns latest persisted and stored block height/index. Persisting
```
[the remark this answers is not in the document]
Comment (AnnaShaleva) [reply pkg/core/interop/context.go]: Fixed.

File: pkg/core/interop/runtime/ext_test.go [hunk 3]
```diff
@@ -738,13 +741,16 @@ func TestSystemRuntimeNotify_HFBasilisk(t *testing.T) {
 	}
 	ctrInv := e.NewInvoker(ctr.Hash, e.Validator)
 
+	// Block 0 is genezis.
```
[the remark this answers is not in the document]
Comment (AnnaShaleva) [reply pkg/core/interop/runtime/ext_test.go]: Fixed.

File: pkg/core/native/notary.go [hunk 4]
```diff
@@ -232,8 +232,8 @@ func (n *Notary) onPayment(ic *interop.Context, args []stackitem.Item) stackitem
 	currentHeight := ic.BlockHeight()
 	deposit := n.GetDepositFor(ic.DAO, to)
 	till := toUint32(additionalParams[1])
-	if till < currentHeight {
-		panic(fmt.Errorf("`till` shouldn't be less then the chain's height %d", currentHeight))
+	if till < currentHeight+2 {
+		panic(fmt.Errorf("`till` shouldn't be less then the chain's height + 1 (%d at min)", currentHeight+2))
```
Comment (roman-khimov) [inline pkg/core/native/notary.go:236]: s/then/than/ while we're here.

**Review by roman-khimov — APPROVED** [review]

File: pkg/core/native/notary.go (hunk 4 above)
Comment (AnnaShaleva) [reply pkg/core/native/notary.go]: And not only here...

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**